homeassistant/components/version/sensor.py

```diff
@@ -1,156 +1,120 @@
+from homeassistant.helpers.device_registry import DeviceEntryType
+from homeassistant.helpers.entity import DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.typing import ConfigType, DiscoveryInfoType, StateType
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import (
+    ATTR_SOURCE,
+    CONF_BETA,
+    CONF_IMAGE,
+    CONF_SOURCE,
+    DEFAULT_BETA,
+    DEFAULT_IMAGE,
+    DEFAULT_NAME,
+    DEFAULT_SOURCE,
+    DOMAIN,
+    HOME_ASSISTANT,
+    VALID_IMAGES,
+    VALID_SOURCES,
+)
+from .coordinator import VersionDataUpdateCoordinator
+
+PLATFORM_SCHEMA: Final[Schema] = SENSOR_PLATFORM_SCHEMA.extend(
     {
-        vol.Optional(CONF_BETA, default=False): cv.boolean,
-        vol.Optional(CONF_IMAGE, default=DEFAULT_IMAGE): vol.In(ALL_IMAGES),
-        vol.Optional(CONF_NAME, default=""): cv.string,
-        vol.Optional(CONF_SOURCE, default=DEFAULT_SOURCE): vol.In(ALL_SOURCES),
+        vol.Optional(CONF_BETA, default=DEFAULT_BETA): cv.boolean,
+        vol.Optional(CONF_IMAGE, default=DEFAULT_IMAGE): vol.In(VALID_IMAGES),
+        vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+        vol.Optional(CONF_SOURCE, default=DEFAULT_SOURCE): vol.In(VALID_SOURCES),
     }
 )
 
-_LOGGER: logging.Logger = logging.getLogger(__name__)
-
-
-async def async_setup_platform(hass, config, async_add_entities, discovery_info=None):
-    """Set up the Version sensor platform."""
-
-    beta = config.get(CONF_BETA)
-    image = config.get(CONF_IMAGE)
-    name = config.get(CONF_NAME)
-    source = config.get(CONF_SOURCE)
-
-    channel = HaVersionChannel.BETA if beta else HaVersionChannel.STABLE
-    session = async_get_clientsession(hass)
-
-    if source in HA_VERSION_SOURCES:
-        source = HaVersionSource(source)
-    elif source == "hassio":
-        source = HaVersionSource.SUPERVISOR
-    elif source == "docker":
-        source = HaVersionSource.CONTAINER
-
-    if (
-        source == HaVersionSource.CONTAINER
-        and image is not None
-        and image != DEFAULT_IMAGE
-    ):
-        image = f"{image}-homeassistant"
-
-    if not (name := config.get(CONF_NAME)):
-        if source == HaVersionSource.LOCAL:
-            name = DEFAULT_NAME_LOCAL
-        else:
-            name = DEFAULT_NAME_LATEST
-
-    async_add_entities(
-        [
-            VersionSensor(
-                VersionData(
-                    HaVersion(
-                        session=session, source=source, image=image, channel=channel
-                    )
-                ),
-                SensorEntityDescription(key=source, name=name),
-            )
-        ],
-        True,
+
+async def async_setup_platform(
+    hass: HomeAssistant,
+    config: ConfigType,
+    async_add_entities: AddEntitiesCallback,
+    discovery_info: DiscoveryInfoType | None = None,
+) -> None:
+    """Set up the legacy version sensor platform."""
```

Comment:
And add it here...

```suggestion
    """Set up the legacy version sensor platform."""
    LOGGER.warning(
        "Configuration of the Version platform in YAML is deprecated and will be "
        "removed in Home Assistant 2022.4; Your existing configuration "
        "has been imported into the UI automatically and can be safely removed "
        "from your configuration.yaml file"
    )
```

This is more how we handle this in general :)
(Both work though 🤷 )

Reply:
That depends on who reviews it 🤷‍♂️

Reply:
Honestly, never seen it being done in the config flow before.

Reply:
<https://github.com/home-assistant/core/pull/46677/files#diff-b2eb3cdf83d0d28d1efac13da0a16c4f5b81917c90caf278da234a26719d02deR109>
<https://github.com/home-assistant/core/pull/53938/files#diff-c6a34945c92acd3a2d0c2ac8516c50341e8aa906cc3fdfb4bd81ee3cad1a5a63R64>
<https://github.com/home-assistant/core/pull/52199/files#diff-ea9255de1cb2d5d5a53771046f2c99ea2c5518d1bc53d51f697b89f247892de4R194>

Reply:
🤷  My review remains the same. I think that is a better location and a more fitting/more end-user-friendly message.